resources/type-system/inference.md

```diff
@@ -474,42 +462,78 @@ replaced with `Null`, and every covariant occurrence of `Ti` replaced with
   from any of the `Bi`:
   - The least closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, [Un+1 xn+1, ..., Um xm])` where:
-    - `U` is the least closure of `T` with respect to `L` 
-    - `Ui` is the greatest closure of `Ti` with respect to `L` 
+    - `U` is the least closure of `T` with respect to `L`
+    - `Ui` is the greatest closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
   - The greatest closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, [Un+1 xn+1, ..., Um xm])` where:
-    - `U` is the greatest closure of `T` with respect to `L` 
-    - `Ui` is the least closure of `Ti` with respect to `L` 
+    - `U` is the greatest closure of `T` with respect to `L`
+    - `Ui` is the least closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
 - if `S` is `T Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn,
-  {Tn+1 xn+1, ..., Tm xm})` and `L` does not contain any free type variables
+  {Tn+1 xn+1, ..., Tm xm})` and  
+  `L` does not contain any free type variables
   from any of the `Bi`:
   - The least closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, {Un+1 xn+1, ..., Um xm})` where:
-    - `U` is the least closure of `T` with respect to `L` 
-    - `Ui` is the greatest closure of `Ti` with respect to `L` 
+    - `U` is the least closure of `T` with respect to `L`
+    - `Ui` is the greatest closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
   - The greatest closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, {Un+1 xn+1, ..., Um xm})` where:
-    - `U` is the greatest closure of `T` with respect to `L` 
-    - `Ui` is the least closure of `Ti` with respect to `L` 
+    - `U` is the greatest closure of `T` with respect to `L`
+    - `Ui` is the least closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
+- if `S` is `T Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn,
+    [Tn+1 xn+1, ..., Tm xm])` or `T Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn,
+{Tn+1 xn+1, ..., Tm xm})`and `L` contains any free type variables
+  from any of the `Bi`:
```

Comment:
Does this rule subsume the first (already existing) rule for function types above, also talking about "contains any free type variables"? It seems that the existing rule was not updated for null safety and still uses `Null` and `Object`.

Reply:
Right, line 455-460 looks like it is now subsumed by 491-496 (so it doesn't matter that 455-460 isn't updated if it's going away).

Reply:
Right, sorry, missed deleting the old definition.  Fixed.